xxhash.go

```diff
@@ -33,19 +33,31 @@ type Digest struct {
 	n     int // how much of mem is used
 }
 
-// New creates a new Digest that computes the 64-bit xxHash algorithm.
+// New creates a new Digest with a zero seed.
 func New() *Digest {
+	return NewWithSeed(0)
+}
+
+// NewWithSeed creates a new Digest with the given seed.
+func NewWithSeed(seed uint64) *Digest {
 	var d Digest
-	d.Reset()
+	d.ResetWithSeed(seed)
 	return &d
 }
 
 // Reset clears the Digest's state so that it can be reused.
+// It uses a seed value of zero.
 func (d *Digest) Reset() {
-	d.v1 = primes[0] + prime2
-	d.v2 = prime2
-	d.v3 = 0
-	d.v4 = -primes[0]
+	d.ResetWithSeed(0)
+}
+
+// ResetWithSeed clears the Digest's state so that it can be reused.
+// It uses the given seed to initialize the state.
+func (d *Digest) ResetWithSeed(seed uint64) {
+	d.v1 = seed + primes[0] + prime2
```

Comment:
Why `primes[0]` instead of `prime1`? For consistency it seems like we should either use `prime1`, `prime2` or `primes[0]`, `primes[1]`.

Reply:
Ah, good catch. I couldn't use the constant here before because I needed to force uint64 arithmetic rather than constant arithmetic. That is, in the old code if I had changed `primes[0]` to `prime1` that would give a compilation error like

    prime1 + prime2 (constant 25430182151971714518 of type uint64) overflows uint64

But in the new code with `seed` this workaround is no longer needed.